packages/components/src/components/FileUploader/FileUploader.tsx

```diff
@@ -0,0 +1,140 @@
+import { useDown } from "@localyze-pluto/theme";
+import React, { ReactElement } from "react";
+import { Box } from "../../primitives/Box";
+import { ButtonProps } from "../Button";
+import { HelpText } from "../HelpText";
+import { FileUploaderButtonProps } from "./FileUploaderButton";
+import { RemoveButton } from "./RemoveButton";
+import { CancelUploadButton } from "./CancelUploadButton";
+import { getStatus } from "./utils/getStatus";
+import { FileUploaderTitle } from "./FileUploaderTitle";
+import { FileUploaderDescription } from "./FileUploaderDescription";
+import { FileUploaderProgressBar } from "./FileUploaderProgressBar";
+import { FileUploaderIcon } from "./FileUploaderIcon";
+
+export interface FileUploaderProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /** The title of the file */
+  label: string;
+
+  /** The call to action button */
+  children: ReactElement<ButtonProps | FileUploaderButtonProps>;
+
+  /** The max size of a file */
+  maxFileSize?: string;
+
+  /** The file name */
+  fileName?: string;
+
+  /** The file url when the file is uploaded */
+  fileUrl?: string;
+
+  /** The size of the file */
+  fileSize?: string;
+
+  /** An error message */
+  errorMessage?: string;
+
+  /** The upload progress */
+  progress?: number;
+
+  /** Handles cancel upload  */
+  onCancel?: () => void;
+
+  /** Handles remove file */
+  onRemove?: () => void;
+}
+
+/** Visual component to display status of a file upload */
+const FileUploader = React.forwardRef<HTMLDivElement, FileUploaderProps>(
+  (
+    {
+      fileName,
+      fileUrl,
+      fileSize,
+      maxFileSize,
+      errorMessage,
+      label,
+      children,
+      progress = 0,
+      onCancel,
+      onRemove,
+    },
+    ref
+  ) => {
+    const isMobile = useDown("md");
+    const status = getStatus(progress, fileUrl, errorMessage);
+    const isMobileUploading = isMobile && status === "loading";
```

Comment:
This condition allows us to more clearly know when to render the mobile version of the upload since the structure is slightly different than all the other versions.

Mobile upload:
<img width="408" alt="Screenshot 2022-11-09 at 11 08 26" src="https://user-images.githubusercontent.com/55699538/200812778-0c7addd3-d1c5-4b6a-a96a-9637745f1e77.png">

Other mobile states:
<img width="368" alt="Screenshot 2022-11-09 at 12 00 02" src="https://user-images.githubusercontent.com/55699538/200817519-bd9a79a0-147f-4a22-a87b-21fdfa586f0c.png">